File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.cpp
class Solution {
private:
    vector<string> findDistinct(vector<string> &arr){
        vector<string> ans;
        unordered_map<string,int> freq;
        for(auto &s: arr){
            freq[s]++;
        }
        for(auto &s: arr){
            if(freq[s]==1){
                ans.push_back(s);
            }
        }
        return ans;
    }
public:
    string kthDistinct(vector<string>& arr, int k) {
        vector<string> res = findDistinct(arr);
        if(k>0 && k<=res.size()){
            return res[k-1];
        }
        return "";
    }
};

File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        freq={}
        for c in arr:
            if c in freq:
                freq[c]+=1
            else:
                freq[c]=1
        for c in arr:
            if c in freq:
                if freq[c]==1:
                    k-=1
                if k==0:
                    return c
        else:
            return ""

            
            